Tests/Unit/Code/ScriptTagBuilderTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Tests\Unit\Code;
+
+use Brotkrueml\MatomoIntegration\Code\ScriptTagBuilder;
+use Brotkrueml\MatomoIntegration\Event\EnrichScriptTagEvent;
+use PHPUnit\Framework\MockObject\Stub;
+use PHPUnit\Framework\TestCase;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Core\Site\Entity\Site;
+
+/**
+ * This test check if the ScriptTagBuilder works as expected
+ */
+final class ScriptTagBuilderTest extends TestCase
+{
+    /**
+     * @var Stub|Site
+     */
+    private $siteStub;
+
+    /**
+     * @var Stub|ServerRequestInterface
+     */
+    private $requestStub;
+
+    protected function setUp(): void
+    {
+        $this->siteStub = $this->createStub(Site::class);
+
+        $this->requestStub = $this->createStub(ServerRequestInterface::class);
+        $this->requestStub
+            ->method('getAttribute')
+            ->with('site')
+            ->willReturn($this->siteStub);
+    }
+
+    /**
+     * @test
+     */
+    public function scriptTagWithoutAttributesRendered(): void
+    {
+        $enrichEvent = new EnrichScriptTagEvent($this->requestStub);
```

Comment:
You can avoid initialisiing an event and stub it, look here:
https://github.com/brotkrueml/typo3-matomo-integration/commit/4ac6676e6e7fb4290645a6d68dc459f8564f5db1
Learned it last week ;-)

Reply:
Yes, that is a nice one. Used it sometimes.